src/ingredient/ingredient.repository.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { IngredientDto } from './dto';
+
+@Injectable()
+export class IngredientRepository {
+  constructor(private readonly prismaService: PrismaService) {}
+
+  async create(dto: IngredientDto) {
+    return this.prismaService.ingredient.create({ data: dto });
+  }
+
+  async findById(id: string) {
+    return this.prismaService.ingredient.findFirst({ where: { id } });
+  }
+
+  async findAll(ids?: string[]) {
```

Comment:
does not make sense, findAll should return all, not allow filtering by ids. name it findMany probably

Reply:
the best thing would be to name it findByIds(), but that does not allow getting all ingredients. But I am not sure, if you need to get all ingredients.